File: t09.cpp
// Write your code here
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main()
{
  int num = 0;
  cin >> num;
  for(;num>0;num--){
    int dim;
    cin >> dim;
    int** a = new int*[dim];
    for(int i = 0; i < dim; i++){
      a[i] = new int[dim];
      for(int j=0;j<dim;j++){
        cin >> a[i][j];
      }
    }

    // for(int i = 0; i < dim; i++){
    //   for(int j=0;j<dim;j++){
    //     cout<<a[i][j];
    //   }
    //   cout<<endl;
    // }
    int count=0;
    for(int i = 0; i < dim; i++){
      for(int j = 0; j < dim; j++){
        int val = a[i][j];
        for(int k = i; k < dim; k++){
          for(int l = j; l < dim; l++){
            if(val > a[k][l])
              count++;
          }
        }
      }
    }
    cout<<count<<endl;

    for(int i=0;i<dim;i++)
      delete[] a[i];
    delete[] a;
    
  }
}

File: t09.py
if __name__ == '__main__':
  num = input()
  for n in range(int(num)):
    dim = int(input())
    a=[]
    for d in range(dim):
      # i1 = input().split()
      i1 = list(map(int, input().split()))
      a=a+[i1]
    # print(a)
    # print(a[0][0])
    # print(a[1][1])
    count = 0
    for i in range(dim):
      for j in range(dim):
        for k in range(i,dim):
          for l in range(j,dim):
            if(a[i][j]>a[k][l]):
              count+=1
    print(count)
  exit()
